File: ExampleApp/app/src/main/java/com/motiv/example/UserWithReferences.java
package com.motiv.example;

import androidx.room.*;

public class UserWithReferences {

    @Embedded private com.motiv.example.User user;

    @Relation(entity = com.motiv.example.Links.class, parentColumn = "linksId", entityColumn = "id")
    public java.util.List<com.motiv.example.Links> links;

    public User getUser() {
        if (links.size() > 0) {
            user.setLinks(links.get(0));
        }

        return user;
    }

    public void setUser(User user) {
        this.user = user;
    }
}

File: ExampleApp/app/src/main/java/com/motiv/example/UserWithReferences.kt
 
package com.motiv.example
import androidx.room.*
import kotlin.collections.List

public class UserWithReferences {

    @Embedded
    private lateinit var user: com.motiv.example.User

    @Relation(entity = com.motiv.example.Links::class, parentColumn = "linksId", entityColumn = "id")

    public var links: List<com.motiv.example.Links> = arrayListOf()

    fun getUser(): User {
        if (links.size > 0) {
            user.setLinks(links.get(0))
        }

        return user
    } public fun setUser(user: User) { this.user = user; }
}
